central/graphql/resolvers/images.go

```diff
@@ -190,24 +191,35 @@ func (resolver *imageResolver) topVulnV2(ctx context.Context, args RawQuery) (Vu
 	return &cVEResolver{root: resolver.root, data: vulns[0]}, nil
 }
 
+func (resolver *imageResolver) vulnQueryScoping(ctx context.Context, query string) (context.Context, string) {
+	ret := search.AddRawQueriesAsConjunction(query, resolver.getImageRawQuery())
+
+	// if postgres is enabled then we should only have to add the id as conjunction and not add scoping
+	if !features.PostgresDatastore.Enabled() {
+		ctx = scoped.Context(ctx, scoped.Scope{
+			Level: v1.SearchCategory_IMAGES,
+			ID:    resolver.data.GetId(),
+		})
+		ctx = distroctx.Context(ctx, resolver.data.GetScan().GetOperatingSystem())
+	}
+
+	return ctx, ret
+}
+
 // Vulns returns all of the vulnerabilities in the image.
-func (resolver *imageResolver) Vulns(ctx context.Context, args PaginatedQuery) ([]VulnerabilityResolver, error) {
-	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.Images, "Vulnerabilities")
+func (resolver *imageResolver) Vulns(ctx context.Context, args PaginatedQuery) ([]ImageVulnerabilityResolver, error) {
```

Comment:
Let's not change the current sub-resolver since it is tightly coupled to type on FE. Instead, add a new a sub-resolver `Vulnerabilities`